config/config.go

```diff
@@ -40,6 +40,10 @@ const (
 	MaxLogFileSize = 4096 // MB
 	// DefTxnTotalSizeLimit is the default value of TxnTxnTotalSizeLimit.
 	DefTxnTotalSizeLimit = 100 * 1024 * 1024
+	// DefMaxIndexLength is the maximum index length. This value is consistent with MySQL.
+	DefMaxIndexLength = 3072
+	// DefTiDBMaxIndexLength is the maximum index length for TiDB v3.0.7 and previous version.
+	DefTiDBMaxIndexLength = 3072 * 4
```

Comment:
In the old calculation method, the index length was not in bytes, but in characters.
That means if we set `max-index-length` to 12288 (3072 * 4), the following statement is valid in current PR but it is invalid in TiDB v3.0.7-.
```sql
create table t12 (a varchar(12288) character set ascii primary key);
```
Is it acceptable?

Reply:
It is acceptable. 
We want to make the following statement is valid in current TiDB like in TiDB v3.0.7-.
```
create table t12 (a varchar(3072) character set ascii primary key);
```